### PR TITLE
Support floating values in exp and nbf claims in io.jwt.verify_decode builtin

### LINKED ISSUE #1296: io.jwt.decode_and_verify does not support non-integer time claims
# Expected Behavior
[rfc7519](https://tools.ietf.org/html/rfc7519#page-10) states that a NumericDate can be a non-integer value (at least, that's my read of it).  The `exp`, `nbf` and`iat` fields are NumericDates. I expect `decode_verify` to support verifying fractional seconds since the epoch for those claims.

If the jwt library doing the validation only supports integer values, I suppose we can get by with taking the floor of the `exp`, and the ceiling of the `nbf` to ensure we're not loosening the claim.

# Actual Behavior
Calling `decode_and_verify` with an otherwise valid jwt containing a non-integer date fails with error: 
`eval_internal_error: io.jwt.decode_verify: parsing 'exp' JWT claim: strconv.ParseInt: parsing "1553722861.6784303": invalid syntax"`

# Steps to Reproduce the Problem

  1. Create a valid JWT with an `nbf` or `exp` claim containing a non-integer numeric value. 
  2. Invoke `io.jwt.decode_verify` with it
 
The following rego playground demonstrates the issue: https://play.openpolicyagent.org/p/4MnaqO4Wx3

This one shows it working with an integer value:
https://play.openpolicyagent.org/p/YjbCHlFh5u

# Additional Information

The python `jwt` library, as well as [jwt.io](https://jwt.io) seem to support non-integer dates.

### PR DESCRIPTION
**Pull request by GBrawl**
Fixes #1296

### CODE REVIEW

**Comment by tsandall** [comment]
@GBrawl it would be good to include a test case that demonstrates that floating point values are supported now.

**Comment by GBrawl** [comment]
Included a couple of simple tests for nbf and exp claim

File: topdown/tokens_test.go [hunk 1]
```diff
@@ -1719,6 +1719,33 @@ func TestTopDownJWTDecodeVerify(t *testing.T) {
...
+		},
+		{
+			"hs256-float-exp-expired",
+			`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMwMDAuMiwiaXNzIjoieHh4In0.XUen7GtDmICV3O1ngsoO-tQrjrXtOgJI06oGW0nQSIM`,
+			`{"secret": "secret", "time": 4000000000000.1}`,
+			false,
+			`{}`,
+			`{}`,
+			"",
+		},
```
Comment (patrick-east) [inline topdown/tokens_test.go:1748]: Could you add in a couple more test cases, I'd like to see:

* When they're equal (to ensure we catch any issues with comparing floats, and any regressions if someone messes with the "if" checks and the compare results), for both `exp` and `nbf`
* A negative case for the `nbf` with floats

**Comment by GBrawl** [comment]
I wrote a couple more tests which should make the current behavior more clear, as you can see it is currently still accurate within a second, not sure if that is what we want

**Comment by patrick-east** [comment]
> I wrote a couple more tests which should make the current behavior more clear, as you can see it is currently still accurate within a second, not sure if that is what we want

I'm not sure, I'm guessing that since the spec supports floats we should at least give our best effort to support our highest precision.

I suspect the issue is from:

```golang
 ast.Number(strconv.FormatInt((constraints.time / 1000000000), 10))
```

We truncate the time here to integer seconds, it should probably be a floating point number to keep as much of the precision from the original nanosecond time as possible.

**Comment by GBrawl** [comment]
@patrick-east I tried your fix and it seems to work fine (after casting it to float64), can you take a look at the tests and check if they suffice?

**Comment by GBrawl** [comment]
Rebased and squashed!

**Pull request merged**